dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/Application.cs

```diff
@@ -138,19 +143,27 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
                 throw new ArgumentException($"{nameof(turnContext)}.Activity must have non-null Type.");
             }
 
+            await this._StartLongRunningCall(turnContext, this._OnTurnAsync, cancellationToken);
+        }
+
+        /// <summary>
+        /// Internal method to wrap the logic of handling a bot turn.
+        /// </summary>
+        private async Task _OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
             TypingTimer? timer = null;
 
             try
             {
                 // Start typing timer if configured
-                if (_options.StartTypingTimer)
+                if (this._options.StartTypingTimer)
                 {
-                    timer = new TypingTimer(_typingTimerDelay);
-                    timer.Start(turnContext);
+                    timer = new TypingTimer(this._typingTimerDelay);
+                    _ = timer.Start(turnContext);
                 };
 
                 // Remove @mentions
-                if (_options.RemoveRecipientMention && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase))
+                if (this._options.RemoveRecipientMention && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: could you remove the `this` from `this._options` to remain consistent with the rest of the code

Reply:
Same for the other use cases

Reply:
Removed. I also changed rule *IDE0009 (Add this or Me qualification)* to *IDE0003 (Remove this or Me qualification)* (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009) to keep consistent with existing code.